src/skmultiflow/trees/extremely_fast_decision_tree.py

```diff
@@ -340,21 +340,20 @@ def _sort_instance_into_leaf(self, X, y, weight):
             found_node.parent.set_child(found_node.parent_branch, leaf_node)
             self._active_leaf_node_cnt += 1
 
-        if isinstance(leaf_node, LearningNode):
+        if not isinstance(leaf_node, SplitNode):
```

Comment:
If we add a new type of Node, then it will pass the condition and we won't find necessary the learn() method. 
I'd keep it like before.

Reply:
The new `EFDTSplitNode` extends the `LearningNodeMC` which in place implements the `learn_one` method.

In this case, we need to change `EFDTSplitNode` to explicitly implement `learn_one` and not inherit it.

Does that sound good @garawalid @smastelini ?

Reply:
I agree with you both. I changed the test because it would trigger a training in the split node in this new organization. But if we remove the inheritance from `LearningNodeMC` we can keep the previous test, as better depicts what is really happening.

Reply:
In commit [ff02f6](https://github.com/scikit-multiflow/scikit-multiflow/pull/271/commits/ff02f65e5ec5e80bcfe17dfbe19b03b2ea13813e) I made the `EFDTSplitNode` implement the needed methods without inheriting from `LearningNodeMC`. Then I reverted the check as it was previously, as suggested by @garawalid.

Reply:
@jacobmontiel I agree with you!
@smastelini that's great!